codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -109,10 +110,12 @@ protected void encode(ChannelHandlerContext ctx, Object msg, List<Object> out) t
         //     ch.write(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
         //
         // See https://github.com/netty/netty/issues/2983 for more information.
-
-        if (msg instanceof ByteBuf && !((ByteBuf) msg).isReadable()) {
-            out.add(EMPTY_BUFFER);
-            return;
+        if (msg instanceof ByteBuf) {
+            final ByteBuf emptyBufPotential = (ByteBuf) msg;
```

Comment:
`potentialEmptyBuf`? That would be similar to `potentialDup` [here](https://github.com/netty/netty/blob/4.1/codec-http2/src/test/java/io/netty/handler/codec/http2/HashCollisionTest.java#L148).

Reply:
variable name updated to be more consistent.